lib/mixlib/versioning/format/rubygems.rb

```diff
@@ -41,23 +45,56 @@ class Format
       #
       # @author Seth Chisamore (<schisamo@chef.io>)
       # @author Christopher Maier (<cm@chef.io>)
+      # @author Ryan Hass (<rhass@chef.io>)
       class Rubygems < Format
-        RUBYGEMS_REGEX = /^(\d+)\.(\d+)\.(\d+)(?:\.([[:alnum:]]+(?:\.[[:alnum:]]+)?))?(?:\-(\d+))?$/
-
         # @see Format#parse
         def parse(version_string)
-          match = version_string.match(RUBYGEMS_REGEX) rescue nil
+          # Pure regex along will never match partial versions correctly in the
+          # expected groupings. This is becausel partial versions shift the
+          # grouping values and break the mappings to for each appropriate value.
+          rubygems_version = version_string.scan(/[0-9]+|[a-z]+/i).map do |s|
+            /^\d+$/ =~ s ? s.to_i : s
+          end rescue nil
+
+          # Each value for MAJOR.MINOR.PATCH must be a integer.
+          unless rubygems_version && rubygems_version[0..2].map {|v| v.is_a?(Integer) }.all?
```

Comment:
Do we have a test for an empty string? If not, would be easy to add one. We could also enforce type to ensure that we aren't passed nil or a non-valid type (like Float).